lib/tasks/after_tasks.rake

```diff
@@ -329,22 +329,27 @@ namespace :After do
 
     return unless Rails.env.staging? || Rails.env.production?
 
-    Collection.where.not(icon_file_name: nil).find_each do |collection|
-      image = collection.icon_file_name
-      ext = File.extname(image)
-      image_original = "original#{ext}"
-
-      # Collection icons are co-mingled in production and staging...
-      icon_url = "https://s3.amazonaws.com/otw-ao3-icons/collections/icons/#{collection.id}/#{image_original}"
-      begin
-        collection.icon.attach(io: URI.parse(icon_url).open,
-                               filename: image_original,
-                               content_type: collection.icon_content_type)
-      rescue StandardError => e
-        puts "Error '#{e}' copying #{icon_url}"
+    Collection.find_in_batches.with_index do |batch, index|
+      batch.each do |collection|
+        next if collection.icon_file_name.blank?
+
+        image = collection.icon_file_name
+        ext = File.extname(image)
+        image_original = "original#{ext}"
+
+        # Collection icons are co-mingled in production and staging...
+        icon_url = "https://s3.amazonaws.com/otw-ao3-icons/collections/icons/#{collection.id}/#{image_original}"
+        begin
+          collection.icon.attach(io: URI.parse(icon_url).open,
+                                 filename: image_original,
+                                 content_type: collection.icon_content_type)
+        rescue StandardError => e
+          puts "Error '#{e}' copying #{icon_url}"
+        end
       end
 
-      puts "Finished up to ID #{collection.id}" if collection.id.modulo(100).zero?
+      puts "Finished batch #{index + 1}"
```

Comment:
I think because you sleep right after, it would good to have a `$stdout.flush` so that this definitely shows up in the output immediately.

And if you want to be fancy you can do the math for the total batch number that the other `find_in_batches` tasks do, but it's definitely not necessary.

Reply:
I would not want to do the count for Pseud, it's large enough that I want to touch the table as little as possible (even things that are relatively safe)